roles/ldap/files/ssh_ldap_wrapper.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+"""
+Usage: ssh_ldap_wrapper.py <user>
+
+Custom ssh-ldap-wrapper script.
+Fetches public keys from LDAP using default ssh-ldap-helper and
+Filters the public keys by dropping unsupported key types or short key sizes considered weak.
+We accept fixed size ed25519 keys and >= 4096 bits rsa keys.
+
+Admin users will be sourced from local credentials. This ensures the system will be maintainable in case of a lost connection to the ldap.
+
+Refactored from a original in bash, which became too obfustcated.
+"""
+
+import logging
+import os.path
+import subprocess
+import sys
+import sshpubkeys
+
+
+class UserKeys(object):
+    """
+    Class holding information about a user and her/his keys.
+    """
+    # The gid of the admin group.
+    admin_gid = 20000
+
+    rsa_key_size = 4096
+    ssh_ldap_helper = '/usr/libexec/openssh/ssh-ldap-helper'
+
+    def __init__(self, user: str):
+        self.user = user
+        if self.is_admin():
+            self.keys = self.local_keys
+        else:
+            self.keys = self.ldap_keys
+
+    def is_admin(self):
+        """
+        Returns:
+            bool: whether the user is an admin.
+        """
+        try:
+            gid = subprocess.run(
+                ['id', '-g', self.user],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                check=True).stdout
+        except subprocess.CalledProcessError as err:
+            logging.error(err)
+            return False
+
+        return int(gid) == self.admin_gid
+
+    def is_ok(self, key: str):
+        """
+        Args:
+            key (str): the ssh key to check.
+        Returns:
+            bool: is the key ok or not.
+        """
+        if key == '':
+            return False
+
+        ssh_key = sshpubkeys.SSHKey(key)
+        try:
+            ssh_key.parse()
+        except sshpubkeys.InvalidKeyError as err:
+            logging.error("Invalid key: {}".format(err))
+            return False
+        except NotImplementedError as err:
+            logging.error("Invalid key type: {}".format(err))
+            return False
+        if ssh_key.key_type == b'ssh-rsa' and ssh_key.bits < self.rsa_key_size:
+            logging.error(
```

Comment:
The check for ```if ssh_key.key_type == b'ssh-rsa'``` is done twice. I'd suggest to do [pseudo code]:
```
if ssh_key.key_type == b'ssh-rsa':
     if ssh_key.bits < self.rsa_key_size:
          ERROR
     else:
          OK
elif ssh_key.key_type == b'ed25519':
      OK
else:
      ERROR
```

Reply:
Thanks. I considered this, but in the end preferred the double check over a nested if statement. I suppose it's a matter of taste.